### DISCUSSION TITLE
Is it good to modify/edit the `ui/components` ?

### DISCUSSION THREAD

**Discussion by JUDE-EQ** [shadcn-ui/ui#3784] [Q&A] [answered]
im doing a lot of modifications at `card.tsx` and `carousel.tsx` component, and i'm wondering if there's an update, then i update the package, would it overwrite my modifications? 

sorry for a noob question but this is my first time using shadcn

**Comment by alvinsjoy** [comment, marked as answer]
Package updates only affect the `package.json` or `package-lock.json`. But if you are running the npx script again (the one  given in the docs under the CLI option Eg: `npx shadcn-ui@latest add carousel
`) your modifications would be overwritten.

You can update just the packages by running `npm update <package_name>` which does not overwrite the ui configs.

**Reply by JUDE-EQ** [reply]
cool, thanks!

**Comment by ChiuMungZitAlexander** [comment]
I don't think so. In my opinion, it's better to create shared components based on generated components. Avoid editting components created by shadcn.
